components/SponsoredFeedsTableWithData.tsx

```diff
@@ -22,56 +20,82 @@ interface SponsoredFeedsTableProps {
 /**
  * Helper functions
  */
+// Convert time_difference (seconds) to human readable format
+const formatTimeUnit = (seconds: number): { value: number; unit: string } => {
```

Comment:
You can use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DurationFormat for this.  It's a bit overkill for our needs as we don't plan to localize pretty much ever, certainly not any time soon, but in general it's probably wise to stick with standards whenever possible.